tools/project_manager/autorelease.inc

```diff
@@ -563,33 +566,37 @@ function AP_setup( $round )
 
 function AP_teardown( $round )
 {
-    mysqli_query(DPDatabase::get_connection(), "
-        DROP TABLE active_page_counts
-    ") or die(DPDatabase::log_error());
+    $sql = "DROP TABLE active_page_counts";
+    DPDatabase::query($sql);
 }
 
 function AP_add_project( $round, $projectid )
 {
-    mysqli_query(DPDatabase::get_connection(), "
+    $sql = sprintf("
         INSERT INTO active_page_counts
         SELECT
-            '$projectid',
-            SUM( state != '{$round->page_save_state}' ) as pages
-        FROM $projectid
-    ") or die(DPDatabase::log_error());
+            '%s',
+            SUM( state != '%s' ) as pages
+        FROM %s",
+        DPDatabase::escape($projectid),
+        DPDatabase::escape($round->page_save_state),
+        DPDatabase::escape($projectid));
+    DPDatabase::query($sql);
 }
 
 function AP_evaluate_criteria( $round, $cooked_project_selector, $release_criterion )
 {
     // Get the criterion-evaluation environment
     // (values for 'projects', 'pages')
-    $res = mysqli_query(DPDatabase::get_connection(), "
+    $sql = sprintf("
         SELECT
-            SUM(projects.state='{$round->project_available_state}') as projects,
+            SUM(projects.state='%s') as projects,
             SUM(active_page_counts.pages) as pages
         FROM projects NATURAL JOIN active_page_counts
-        WHERE $cooked_project_selector
-    ") or die(DPDatabase::log_error());
+        WHERE %s",
```

Comment:
Leave `$cooked_project_selector` in the string since it is built elsewhere and can't be escaped.

Reply:
The only reason I pulled all of these out was in case there was a % in the text that would be expanded before being passed to sprintf. Then it could run into an issue trying to sprintf the string. However, in these cases I think we are ok.